File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> v = new ArrayList<>();
        recursive(nums, ans, 0, v);
        return ans;
    }

    private void recursive(int[] nums, List<List<Integer>> ans, int i, List<Integer> v) {
        if (i == nums.length) {
            ans.add(new ArrayList<>(v));
            return;
        }
        
        recursive(nums, ans, i + 1, v);
        
        v.add(nums[i]);
        recursive(nums, ans, i + 1, v);
        
        v.remove(v.size() - 1); // Backtrack by removing the last element
    }
}

File: 78-subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int n=nums.size();
        vector<vector<int>> ans;
        for(int i=0; i<(1<<n); i++){ // alternate pow(2, n);
            vector<int> v;
            for(int j=0; j<n; j++){
                if((1<<j) & i) v.push_back(nums[j]); //alternate- ((1<<j) & i) !=0 in if condn
            }
            ans.push_back(v);
        }
        return ans;
    }
};